static/js/message_store.js

```diff
@@ -304,12 +304,10 @@ exports.update_messages = function update_messages(events) {
 
     // If a topic was edited, we re-render the whole view to get any propagated edits
     // to be updated
-    if (topic_edited) {
-        if (!changed_narrow) {
-            home_msg_list.rerender();
-            if (current_msg_list === message_list.narrowed) {
-                message_list.narrowed.rerender();
-            }
+    if (topic_edited && !changed_narrow) {
+        home_msg_list.rerender();
+        if (current_msg_list === message_list.narrowed) {
+            message_list.narrowed.rerender();
```

Comment:
Well, it _does_ fix the issue, but I think it introduces a few more (like topic edits not updating in other views). Given that the logic comes strait from b51ea69db03e23fbc612ae301cc83f8f4dcd30b3, I'm not sure if it should be changed..